### PR TITLE
Add cluster_name to proxy kubernetes config

### PR DESCRIPTION
**Pull request by awly**
Cluster name from this field plug all clusters from kubeconfig are
stored on the auth server via heartbeats.
This info will later be used to route k8s requests back to proxies.

Updates https://github.com/gravitational/teleport/issues/3952

### CODE REVIEW

File: lib/srv/regular/sshserver.go [hunk 1]
```diff
@@ -712,6 +720,7 @@ func (s *Server) getServerInfo() (services.Server, error) {
 	}
 	server.SetTTL(s.clock, defaults.ServerAnnounceTTL)
 	server.SetPublicAddr(s.proxyPublicAddr.String())
+	server.SetKubernetesClusters(s.proxyKubenretesClusters)
```
Comment (webvictim) [inline lib/srv/regular/sshserver.go:723]:
```suggestion
	server.SetKubernetesClusters(s.proxyKubernetesClusters)
```

File: lib/srv/regular/sshserver.go [hunk 2]
```diff
@@ -465,6 +466,13 @@ func SetOnHeartbeat(fn func(error)) ServerOption {
 	}
 }
 
+func SetKubernetesClusters(clusters []string) ServerOption {
+	return func(s *Server) error {
+		s.proxyKubenretesClusters = clusters
```
Comment (webvictim) [inline lib/srv/regular/sshserver.go:471]:
```suggestion
		s.proxyKubernetesClusters = clusters
```

File: lib/srv/regular/sshserver.go [hunk 3]
```diff
@@ -86,8 +86,9 @@ type Server struct {
 	proxyMode bool
 	proxyTun  reversetunnel.Server
 
-	advertiseAddr   *utils.NetAddr
-	proxyPublicAddr utils.NetAddr
+	advertiseAddr           *utils.NetAddr
+	proxyPublicAddr         utils.NetAddr
+	proxyKubenretesClusters []string
```
Comment (webvictim) [inline lib/srv/regular/sshserver.go:91]:
```suggestion
	proxyKubernetesClusters []string
```

**Comment by webvictim** [comment]
LGTM (once the conflicts are fixed) - I'll leave it for the others as it's core functionality though.

**Comment by awly** [comment]
retest this please

File: lib/srv/regular/sshserver.go [hunk 4]
```diff
@@ -457,6 +458,13 @@ func SetOnHeartbeat(fn func(error)) ServerOption {
 	}
 }
 
+func SetKubernetesClusters(clusters []string) ServerOption {
```
Comment (russjones) [inline lib/srv/regular/sshserver.go:461]: How is this used on a SSH server?

Comment (awly) [reply lib/srv/regular/sshserver.go:468]: Added a comment.
`lib/srv/regular.Server` manages proxy heartbeats. Heartbeats is how a proxy announces its supported k8s clusters to the rest of the teleport cluster.
Logically, heartbeats are a separate feature from SSH serving, but untangling these would be a noisy refactoring.

Comment (russjones) [reply lib/srv/regular/sshserver.go:468]: Makes sense.

**Review by russjones — APPROVED** [review]

**Pull request merged**